keyboards/mechwild/sugarglider/sugarglider.c

```diff
@@ -0,0 +1,441 @@
+            } else {
+                tap_code(KC_VOLD);
+            }
+        break;
+        case 2:
+            if (clockwise) {
+                tap_code(KC_PGUP);
+            } else {
+                tap_code(KC_PGDN);
+            }
+        break;
+        case 3:
+            if (clockwise) {
+                tap_code(KC_WH_U);
+            } else {
+                tap_code(KC_WH_D);
+            }
+        break;
+    }
+    return true;
+}
+#endif
+
+bool led_update_kb(led_t led_state) {
+    bool res = led_update_user(led_state);
+    if(res) {
+        writePin(B12, led_state.num_lock);     // Updates status LEDs
+        writePin(B13, led_state.caps_lock);    // Updates status LEDs
+        writePin(B14, led_state.scroll_lock);  // Updates status LEDs
+        writePin(C13, !led_state.caps_lock);   // Updates status LEDs, this is the LED on the blackpill itself
+    }
+    return res;
+}
+
+#if defined(OLED_ENABLE) && defined(OLED_DISPLAY_128X64) // Wide OLED Functionality
+oled_rotation_t oled_init_user(oled_rotation_t rotation) {
+    return OLED_ROTATION_180;       // flips the display 180 degrees
+}
+
+bool clear_screen = true;           // used to manage singular screen clears to prevent display glitch
+bool clear_screen_art = true;       // used to manage singular screen clears to prevent display glitch
+static void render_sugarglider_wide(void) {     // Render sugarglider logo
+    static const char PROGMEM sgl_1[] = {0xDE, 0xDE, 0xDE, 0x98, 0x99, 0xDE, 0x9A, 0x9B, 0x9C, 0xDE, 0xDE, 0xDE, 0xDE, 0xDE, 0xCC, 0xCD, 0xCE, 0xCF, 0x00};
+    static const char PROGMEM sgl_2[] = {0x9D, 0x9E, 0xDE, 0x9F, 0xA0, 0xA1, 0xA2, 0xA3, 0xA4, 0xA5, 0xA6, 0xDE, 0xDE, 0xDE, 0xD0, 0xD1, 0xD2, 0xD3, 0xD4, 0x00};
+    static const char PROGMEM sgl_3[] = {0xA7, 0xA8, 0xA9, 0xAA, 0xAB, 0xAC, 0xAD, 0xAE, 0xAF, 0xB0, 0xB1, 0xDE, 0xDE, 0xDE, 0xD5, 0xD6, 0xD7, 0xD8, 0x00};
+    static const char PROGMEM sgl_4[] = {0xDE, 0xB2, 0xDE, 0xB3, 0xB4, 0xB5, 0xB6, 0xDE, 0xDE, 0xB7, 0xDE, 0xDE, 0xDE, 0xDE, 0xDE, 0xD9, 0xDA, 0xDB, 0x00};
+    static const char PROGMEM sgl_5[] = {0xDE, 0xB8, 0xDE, 0xDE, 0xDE, 0xDE, 0xDE, 0xDE, 0xB9, 0x00};
+    static const char PROGMEM sgl_6[] = {0xDE, 0xBA, 0xDE, 0xDE, 0xDE, 0xDE, 0xDE, 0xDE, 0xBB, 0xBC, 0x00};
+    static const char PROGMEM sgl_7[] = {0xDE, 0xBD, 0xBE, 0xDE, 0xDE, 0xBF, 0xDE, 0xDE, 0xC0, 0xC1, 0x00};
+    static const char PROGMEM sgl_8[] = {0xC2, 0xC3, 0xC4, 0xC5, 0xC6, 0xC7, 0xC8, 0xC9, 0xCA, 0xCB, 0x00};
+
+    oled_set_cursor(0,0);
+    oled_write_P(sgl_1, false);
+    oled_set_cursor(0,1);
+    oled_write_P(sgl_2, false);
+    oled_set_cursor(0,2);
+    oled_write_P(sgl_3, false);
+    oled_set_cursor(0,3);
+    oled_write_P(sgl_4, false);
+    oled_set_cursor(0,4);
+    oled_write_P(sgl_5, false);
+    oled_set_cursor(0,5);
+    oled_write_P(sgl_6, false);
+    oled_set_cursor(0,6);
+    oled_write_P(sgl_7, false);
+    oled_set_cursor(0,7);
+    oled_write_P(sgl_8, false);
```

Comment:
shouldn't use set cursor unless absolutely necessary. It can cause issues, and size increases, unwanted refreshing, etc.  Advance page goes to the next line "properly".
```suggestion
    oled_advance_page(true);
    oled_write_P(sgl_2, false);
    oled_advance_page(true);
    oled_write_P(sgl_3, false);
    oled_advance_page(true);
    oled_write_P(sgl_4, false);
    oled_advance_page(true);
    oled_write_P(sgl_5, false);
    oled_advance_page(true);
    oled_write_P(sgl_6, false);
    oled_advance_page(true);
    oled_write_P(sgl_7, false);
    oled_advance_page(true);
    oled_write_P(sgl_8, false);
```

Reply:
These changes just don't show the lines as expected and it breaks the whole display when I try and write over other parts of the display, such as with the dynamic text, layers, etc. It works well if I don't try and write over anything.